sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1156,6 +1202,13 @@ def need_ray_up(
 
         logger.debug(f'Ray up takes {time.time() - start} seconds with '
                      f'{retry_cnt} retries.')
+        if returncode != 0:
+            return self.GangSchedulingStatus.HEAD_FAILED, stdout, stderr, None
+
+        resources = cluster_handle.launched_resources
+        if tpu_utils.is_tpu_vm(resources) and tpu_utils.is_tpu_pod(resources):
+            logger.info('Setting up TPU Pod workers...')
```

Comment:
nit: Why don't we add `f'{style.BRIGHT}` just like L1231?

Reply:
good point! added